Bug Report: Crash when pushing enum or set values in the evalengine

### Overview of the Issue

When an enum or set value is pushed in the evalengine, it can lead to a panic because of a wrong copy paste.

```
panic: runtime error: index out of range [-1] [recovered]
	panic: runtime error: index out of range [-1]
```

### Reproduction Steps

Have a table with an `enum` column that ends up used in the `evalengine`. 

### Binary Version

```sh
-
```


### Operating System and Environment details

```sh
-
```


### Log Fragments

_No response_

--- a/go/vt/vtgate/evalengine/arena.go
+++ b/go/vt/vtgate/evalengine/arena.go
@@ -71,7 +71,7 @@ func (a *Arena) newEvalEnum(raw []byte, values *EnumSetValues) *evalEnum {
 	} else {
 		a.aEnum = append(a.aEnum, evalEnum{})
 	}
-	val := &a.aEnum[len(a.aInt64)-1]
+	val := &a.aEnum[len(a.aEnum)-1]
 	s := string(raw)
 	val.string = s
 	val.value = valueIdx(values, s)
@@ -84,7 +84,7 @@ func (a *Arena) newEvalSet(raw []byte, values *EnumSetValues) *evalSet {
 	} else {
 		a.aSet = append(a.aSet, evalSet{})
 	}
-	val := &a.aSet[len(a.aInt64)-1]
+	val := &a.aSet[len(a.aSet)-1]
 	s := string(raw)
 	val.string = s
 	val.set = evalSetBits(values, s)

--- a/go/vt/vtgate/evalengine/compiler_test.go
+++ b/go/vt/vtgate/evalengine/compiler_test.go
@@ -760,6 +760,20 @@ func TestCompilerSingle(t *testing.T) {
 			expression: `WEEK(timestamp '2024-01-01 10:34:58', 1)`,
 			result:     `INT64(1)`,
 		},
+		{
+			expression: `column0 + 1`,
+			values:     []sqltypes.Value{sqltypes.MakeTrusted(sqltypes.Enum, []byte("foo"))},
+			// Returns 0, as unknown enums evaluate here to -1. We have this test to
+			// exercise the path to push enums onto the stack.
+			result: `FLOAT64(0)`,
+		},
+		{
+			expression: `column0 + 1`,
+			values:     []sqltypes.Value{sqltypes.MakeTrusted(sqltypes.Set, []byte("foo"))},
+			// Returns 1, as unknown sets evaluate here to 0. We have this test to
+			// exercise the path to push sets onto the stack.
+			result: `FLOAT64(1)`,
+		},
 	}
 
 	tz, _ := time.LoadLocation("Europe/Madrid")

